monai/networks/utils.py

```diff
@@ -615,6 +626,109 @@ def convert_to_torchscript(
+            We suggest the `OPT_BATCH` to be the most frequently used input batchsize in your application, default to None.
+        use_trace: whether using `torch.jit.trace` to convert the pytorch model to torchscript model and then convert to
+            a TensorRT engine based torchscript model, default to False.
+        filename_or_obj: if not None, specify a file-like object (has to implement write and flush) or a string containing a
+            file path name to load the TensorRT engine based torchscript model for verifying.
+        verify: whether to verify the input and output of the TensorRT engine based torchscript model.
+        device: the target GPU index to convert and verify the model. If None, use #0 GPU.
+        rtol: the relative tolerance when comparing the outputs between the PyTorch model and TensorRT model.
+        atol: the absolute tolerance when comparing the outputs between the PyTorch model and TensorRT model.
+    """
+
+    if not has_torch_tensorrt:
+        raise ImportError("Cannot find the torch_tensorrt module, please install it before converting.")
+
+    if not torch.cuda.is_available():
+        raise Exception("Cannot find any GPU devices.")
+
+    if not input_shape:
+        raise ValueError("Missing the input shape for model convert.")
+
+    if not dynamic_batchsize:
+        warnings.warn(f"There is no dynamic batch range. The converted model only takes {input_shape} shape input.")
+
+    if (not (dynamic_batchsize is None)) and (len(dynamic_batchsize) != 3):
+        warnings.warn(f"The dynamic batch range sequence should have 3 elements, but got {dynamic_batchsize} elements.")
+
+    device = device if device else 0
+    target_device = torch.device(f"cuda:{device}") if device else torch.device("cuda:0")
+    convert_precision = torch.float32 if precision == "fp32" else torch.half
+    inputs = [torch.rand(ensure_tuple(input_shape)).to(target_device)]
+
+    # convert the torch model, torchscript model and input to target device
+    model = model.eval().to(target_device)
+    ir_model = convert_to_torchscript(model, device=target_device, inputs=inputs, use_trace=use_trace)
+    ir_model.eval().to(target_device)
+
+    def scale_batch_size(input_shape: Sequence[int], scale_num: int):
+        scale_shape = [*input_shape]
+        scale_shape[0] *= scale_num
+        return scale_shape
+
+    # Use the dynamic batchsize range to generate compiler input and compile the model
+    if dynamic_batchsize:
+        min_input_shape = scale_batch_size(input_shape, dynamic_batchsize[0])
+        opt_input_shape = scale_batch_size(input_shape, dynamic_batchsize[1])
+        max_input_shape = scale_batch_size(input_shape, dynamic_batchsize[2])
+    else:
+        min_input_shape = opt_input_shape = max_input_shape = input_shape
+
+    with torch.no_grad():
+        torch_tensorrt.set_device(device)
```

Comment:
ci runners report this issue
```
[2023-03-21T09:11:04.418Z] ======================================================================
[2023-03-21T09:11:04.418Z] FAIL: test_visible_devices (tests.test_set_visible_devices.TestVisibleDevices)
[2023-03-21T09:11:04.418Z] ----------------------------------------------------------------------
[2023-03-21T09:11:04.418Z] Traceback (most recent call last):
[2023-03-21T09:11:04.418Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/tests/test_set_visible_devices.py", line 36, in test_visible_devices
[2023-03-21T09:11:04.418Z]     self.assertEqual(num_gpus_before, num_gpus_after)
[2023-03-21T09:11:04.418Z] AssertionError: 0 != 1
[2023-03-21T09:11:04.418Z] 
[2023-03-21T09:11:04.418Z] ----------------------------------------------------------------------
[2023-03-21T09:11:04.418Z] Ran 12329 tests in 1396.392s
[2023-03-21T09:11:04.418Z] 
[2023-03-21T09:11:04.418Z] FAILED (failures=1, skipped=490)
script returned exit code 1
```
I suspect that this set_device seems to change the system device globally and affects the other unit tests

Reply:
Hi @wyli , 
     I think this is a blocker but need @Nic-Ma  to double confirm. If so, how about I just remove this line and add `currently only support the index 0 GPU` to the `device` doc-string?  This change will make the `trt_export` API only work on GPU0.

Thanks,
Bin

Reply:
or somehow reset the device in the unit tests

Reply:
Hi @wyli ,
     I haven't found any `reset` like API in Torch_TensorRT. But find the `torch.cuda.device` context manager in pytorch, which works like the `torch_tensorrt.set_device()`.  I can change the `torch_tensorrt.set_device(device)` to `with torch.cuda.device(device=device):` and may need your help to run the ci test again to see if this issue can be fixed.

Thanks,
Bin

Reply:
I edited the unit tests and the prints are not consistent in setup and teardown, so I don't think it works...

![Screenshot 2023-03-21 at 13 35 33](https://user-images.githubusercontent.com/831580/226622442-a4548aa7-7fd4-4d0c-9216-68ece14f02c4.png)

Reply:
the root cause is that when importing torch_tensorrt, it'll change torch.cuda.device_count, ignoring the environment variable.

Reply:
Hi @wyli ,

Do you think we should report this behavior to `torch TensorRT` project and ask to change?

Thanks.